hbase-common/src/main/resources/hbase-default.xml

```diff
@@ -1930,4 +1930,21 @@ possible configurations would overwhelm and obscure the important.
       this feature.
     </description>
   </property>
+  <property>
+    <name>hbase.hstore.file.force.archive</name>
+    <value>true</value>
```

Comment:
Should this be false by default? I think for Phoenix like cases you can support per query to make it on? Rather than cluster level? Just asking . Which one wil be better here?

Reply:
I am still not sure if making it false would be better. High level, what we are proposing is for HBase to take care of old scanners that are preventing archival of compacted files and the solution is for HBase to reset scanners in heap and continue operations. Since this solution is gracefully taking care of removing reader references from compacted files without impacting clients, should we not have this enabled by default?

Reply:
What we do here is reverting the optimization done by adding the CompactionDischarger thread way. And we make it default ON. This is wrong IMO.  And we initially discussed abt reset this scanner iff the scanners over the compacted file are not closed after some time period even. But seems we dont do that also.

Reply:
By enabling this, here is what would happen:
1. compaction happens
2. old reader ref would prevent archival of compacted away store files
3. With this config, we would notify ChangedReaderObserver because of which subsequent seek(), next() on scanner would reset scanners and add them in heap and also ensure client scan is not broken
4. next run of discharger chore, if gets 0 refCount on the same file, would archive them

If above steps look good for default behavior, we can keep this as is.

>> And we initially discussed abt reset this scanner iff the scanners over the compacted file are not closed after some time period even. But seems we dont do that also.

Since we later decided to gracefully take care of this problem, we are not breaking the scanners forcefully so that client read is not impacted.

Reply:
If you feel the above mentioned steps look good for default behavior but the config name should be changed to something better to represent them, please let me know, I will update it.

Reply:
Keeping perf issues in mind, let me make this config disabled by default